src/main/java/net/minecraftforge/fml/common/registry/ObjectHolderRef.java

```diff
@@ -99,23 +102,18 @@
 
     public boolean isValid()
     {
-        return isBlock || isItem;
+        return isValid;
     }
     public void apply()
     {
         Object thing;
-        if (isBlock)
+        if (isValid)
         {
-            thing = ForgeRegistries.BLOCKS.getValue(injectedObject);
-            if (thing == Blocks.AIR)
-            {
+            thing = registry.getValue(injectedObject);
+            if (registry.getRegistrySuperType() == Block.class && thing == Blocks.AIR) {
```

Comment:
Hmn, that's a fair call.

In order to make it more generic, I'd have to know what the optional default object for the registry is, and IForgeRegistry<T> doesn't have a method for me to retrieve it.

Would you find it reasonable to add a method to IForgeRegistry to support this, or perhaps use a secondary interface and instanceof check it?